iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergInserts.java

```diff
@@ -183,6 +183,22 @@ public void testInsertOverwriteBucketPartitionedTableThrowsError() {
             testTables.getInsertQuery(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS, target, true)));
   }
 
+  @Test
+  public void testInsertOverwriteWithPartitionEvolutionThrowsError() throws IOException {
```

Comment:
Actually, maybe the merge won't help here either using the current logic, because it won't delete the old data files, just remove the rows using delete files...
So only compaction (https://iceberg.apache.org/docs/latest/maintenance/#compact-data-files) will help - @pvary any thoughts ?